### PR TITLE
make Distributions a weak dependency

### PR DESCRIPTION
**Pull request by oscardssmith**
On 1.9, this saves roughly .5 seconds. We also should do this for chainrules, but I'll leave that for another PR.

Before
```

julia> @time @eval using DiffEqBase
  1.914418 seconds (6.02 M allocations: 432.638 MiB, 6.05% gc time, 16.83% compilation time: <1% of which was recompilation)
```
After
```
julia> @time @eval using DiffEqBase
  1.516961 seconds (4.86 M allocations: 365.564 MiB, 5.58% gc time, 18.51% compilation time: <1% of which was recompilation)
```

### CODE REVIEW

File: src/solve.jl [hunk 1]
```diff
@@ -1082,8 +1082,6 @@ function get_concrete_p(prob, kwargs)
 end
 
 handle_distribution_u0(_u0) = _u0
-handle_distribution_u0(_u0::Distributions.Sampleable) = rand(_u0)
-isdistribution(_u0::Distributions.Sampleable) = true
```
Comment (ChrisRackauckas) [inline src/solve.jl:1086]: this should be in the `else` of the `isdefined`?

Comment (oscardssmith) [reply src/solve.jl:1086]: what do you mean?

Comment (ChrisRackauckas) [reply src/solve.jl:1086]: good point

Comment (ChrisRackauckas) [reply src/solve.jl:1086]: wait no, isn't this breaking for v1.7?

Comment (oscardssmith) [reply src/solve.jl:1086]: no. As described in https://pkgdocs.julialang.org/dev/creating-packages/#Transition-from-normal-dependency-to-extension this PR will make Distributions act as a normal dependency in 1.8 or below (and will load these 2 lines)

Comment (ChrisRackauckas) [reply src/solve.jl:1086]: but the `isdefined` will be false, so it will not include the file?

Comment (oscardssmith) [reply src/solve.jl:1086]: It's `if !isdefined(Base, :get_extension)` (note the not)

Comment (ChrisRackauckas) [reply src/solve.jl:1086]: Then the file is not included in v1.9? In what case is the file included and not included? For it to not break, it needs to be included in all versions, so I don't get how an `if` can even work.

Comment (oscardssmith) [reply src/solve.jl:1086]: On 1.9 it gets included by the
```
[extensions]
DistributionsExt = "Distributions"
```
in the Project.toml which will load `DistributionsExt.jl` if (and only if) `Distributions` is in the user's `Manifest.toml`

Comment (anandijain) [reply src/solve.jl:1086]: where's the docs to extension packages?

**Comment by ChrisRackauckas** [comment]
We should do Measurements and stuff too

**Pull request merged**